### PR TITLE
[sqllab] assign types for visualize flow

### LINKED ISSUE #2430: [sql-lab] table columns created from sql query are not created with correct type
sql-lab-visuallize-table-column-bug

### PR DESCRIPTION
**Pull request by mistercrunch**
Somehow when using the visualize flow, the types were not
assigned at all, creating some bugs downstream. This PR attempts to get
the information required based on what pandas is knows and the types in
the data itself.

Side mission: cleaning up the module level functions into classmethods

closes https://github.com/airbnb/superset/issues/2430#event-1011634216

### CODE REVIEW

File: superset/dataframe.py [hunk 1]
```diff
@@ -19,6 +22,21 @@
...
 
 class SupersetDataFrame(object):
+    # Mapping numpy dtype.char to generic database types
+    type_map = {
+        'b': 'BOOL',  # boolean
+        'i': 'INT',  # (signed) integer
+        'u': 'INT',  # unsigned integer
+        'f': 'FLOAT',  # floating-point
+        'c': 'FLOAT',  # complex-floating point
+        'm': None,  # timedelta
```
Comment (ascott) [inline superset/dataframe.py]: why is the value here `None` but the comment says `timedelta`?

File: superset/dataframe.py [hunk 2]
```diff
@@ -30,6 +48,47 @@ def size(self):
     def data(self):
         return self.__df.to_dict(orient='records')
 
+    @classmethod
+    def db_type(cls, dtype):
```
Comment (ascott) [inline superset/dataframe.py]: shouldn't this be `col_type` rather than `db_type` here?

**Comment by ascott** [comment]
it would be great to add a test here that asserts that when a table is created, the column types are correctly saved as well.

**Comment by mistercrunch** [comment]
That test did actually exists but is failing ... Fixing it now

File: superset/dataframe.py (hunk 1 above)
Comment (mistercrunch) [inline superset/dataframe.py]: I wanted to cover the entire scope of numpy types to be explicit, but it really doesn't map to a classic database type. We could handle / convert those in the future but most likely database drivers will never return that type (though they are supported in numpy...)

File: superset/dataframe.py (hunk 2 above)
Comment (mistercrunch) [inline superset/dataframe.py]: yes to be explicit I should differentiate between col_numpy_type and col_db_type, which I'm converting here...

**Comment by mistercrunch** [comment]
Thanks for the review!

**Pull request merged**